src/antd/InputNumber.tsx

```diff
@@ -41,21 +62,10 @@ export function InputNumber({
     (Array.isArray(numberType) &&
       numberType.length === 1 &&
       numberType.includes("integer"))
-  // TODO: Revist useCallback use - was meant to prevent re-renders
-  const onChange = useCallback(
-    (value: number | string | null) => {
-      if (typeof value === "number") {
-        if (isInteger) {
-          handleChange(path, coerceToInteger(value))
-        } else {
-          handleChange(path, coerceToNumber(value))
-        }
-      } else {
-        handleChange(path, value)
-      }
-    },
-    [handleChange, isInteger, path],
-  )
+
+  const onChange = (value: number | string | null) => {
+    createInputNumberChangeHandler(handleChange, path, isInteger)(value)
```

Comment:
I'd suggest avoiding the IIFE format, as it adds unnecessary complexity. 
```suggestion
     if (typeof value === "number") {
      if (isInteger) {
        handleChange(path, coerceToInteger(value))
      } else {
        handleChange(path, coerceToNumber(value))
      }
    } else {
      handleChange(path, value)
    }
```

Reply:
I think I wanna hang onto it here because it ensures that each invocation of `createInputNumberChangeHandler` is unique to the `value` that's passed in, since it spawns a unique underlying closure. I can leave a comment explaining that though 👍